### ISSUE TITLE
Create UInt64 from UInt32

### ISSUE THREAD

**Issue by iskiselev** [sq/JSIL#855]
JSIL translated code fails on next test case:
```
using System;

public static class Program {
    public static void Main (string[] args)
    {
        ulong l = 0x80000000;
        Console.WriteLine(l);
    }
}
```

It fails with `Error: cannot construct UInt64 from negative number`.
This code was translated to:
```
var $T02 = function () {
    return ($T02 = JSIL.Memoize($asm01.System.UInt64)) ();
  };
    $T01().WriteLine($T02().FromNumber(-2147483648));
```

So, probably we should allow creation of UInt64 from negative numbers with built-in conversion or fix how JSIL translates it.

**Comment by markusjohnsson** [comment]
Yup, FromNumber should probably interpret -2147483648 as a 32 bit int, take the bits from that and treat it as a 32 bit uint. 

See conv_u8: https://github.com/markusjohnsson/cil.js/blob/master/CilJs/Runtime/Runtime.js#L363
